File: client-web/src/app/core/models/order.models.ts
export enum OrderStatus {
  New = 0,
  Received = 1,
  Sorting = 2,
  Washing = 3,
  Drying = 4,
  Ironing = 5,
  Packing = 6,
  ReadyForDelivery = 7,
  Delivered = 8,
  Cancelled = 9,
  OutForDelivery = 10
}

export enum PaymentStatus {
  Pending = 0,
  Partial = 1,
  Paid = 2,
  Refunded = 3
}

export enum OrderChannel {
  WalkIn = 0,
  PickupRequest = 1,
  Express = 2
}

export interface OrderListItem {
  id: string;
  orderNumber: string;
  customerName: string;
  status: OrderStatus;
  totalAmount: number;
  paymentStatus: PaymentStatus;
  createdAtUtc: string;
  promisedByUtc?: string | null;
}

export interface PaginatedList<T> {
  items: T[];
  pageNumber: number;
  totalPages: number;
  totalCount: number;
}

export interface OrderItemDetail {
  id: string;
  garmentName: string;
  serviceName: string;
  quantity: number;
  weightKg?: number | null;
  unitPrice: number;
  lineTotal: number;
  specialInstructions?: string | null;
}

export interface OrderDetail {
  id: string;
  orderNumber: string;
  status: OrderStatus;
  channel: OrderChannel;
  isExpress: boolean;
  subTotal: number;
  discountAmount: number;
  promoCode?: string | null;
  gstAmount: number;
  deliveryCharge: number;
  totalAmount: number;
  paymentStatus: PaymentStatus;
  amountPaid: number;
  createdAtUtc: string;
  promisedByUtc?: string | null;
  deliveredAtUtc?: string | null;
  pickupScheduledAtUtc?: string | null;
  items: OrderItemDetail[];
}

export interface CreateOrderItem {
  garmentId: string;
  serviceId: string;
  quantity: number;
  weightKg?: number;
  specialInstructions?: string;
}

export interface UpdateOrderRequest {
  status?: OrderStatus;
  paymentStatus?: PaymentStatus;
  amountPaid?: number;
  promisedByUtc?: string;
}

export interface CreateOrderRequest {
  customerId: string;
  channel: OrderChannel;
  isExpress: boolean;
  items: CreateOrderItem[];
  preferredPickupAtUtc?: string;
  pickupAddressId?: string;
  promoCode?: string;
}

export interface LoginRequest {
  usernameOrEmail: string;
  password: string;
}

export interface RegisterRequest {
  fullName: string;
  phoneNumber: string;
  email: string;
  password: string;
}

export interface VerifyOtpRequest {
  phoneNumber: string;
  code: string;
}

export interface RegisterResponse {
  userId: string;
  devOtpCode?: string;
}

export interface ForgotPasswordRequest {
  usernameOrEmail: string;
}

export interface ForgotPasswordResponse {
  accountFound: boolean;
  devOtpCode?: string;
}

export interface ResetPasswordRequest {
  usernameOrEmail: string;
  code: string;
  newPassword: string;
}

export interface LoginResponse {
  accessToken: string;
  refreshToken: string;
  expiresAtUtc: string;
  userId: string;
  fullName: string;
  role: string;
}

File: LaundryMgmt.Mobile/Models/OrderModels.cs
namespace LaundryMgmt.Mobile.Models;

public enum OrderStatus
{
    New = 0, Received = 1, Sorting = 2, Washing = 3, Drying = 4,
    Ironing = 5, Packing = 6, ReadyForDelivery = 7, Delivered = 8, Cancelled = 9, OutForDelivery = 10
}

public enum PaymentStatus
{
    Pending = 0, Partial = 1, Paid = 2, Refunded = 3
}

public enum OrderChannel
{
    WalkIn = 0, PickupRequest = 1, Express = 2
}

public record OrderListItem(
    Guid Id, string OrderNumber, string CustomerName, OrderStatus Status,
    decimal TotalAmount, PaymentStatus PaymentStatus, DateTimeOffset CreatedAtUtc,
    DateTimeOffset? PromisedByUtc);

public record CreateOrderItemRequest(Guid GarmentId, Guid ServiceId, int Quantity, decimal? WeightKg, string? SpecialInstructions);

public record CreateOrderRequest(
    Guid CustomerId, OrderChannel Channel, bool IsExpress, List<CreateOrderItemRequest> Items,
    DateTimeOffset? PreferredPickupAtUtc = null, Guid? PickupAddressId = null, string? PromoCode = null);

public record OrderItemDetailDto(
    Guid Id, string GarmentName, Guid ServiceId, string ServiceName, int Quantity, decimal? WeightKg,
    decimal UnitPrice, decimal LineTotal, string? SpecialInstructions);

public record UpdateOrderRequest(OrderStatus? Status, PaymentStatus? PaymentStatus, decimal? AmountPaid, DateTimeOffset? PromisedByUtc);

public record OrderDetailDto(
    Guid Id, string OrderNumber, OrderStatus Status, OrderChannel Channel, bool IsExpress,
    decimal SubTotal, decimal DiscountAmount, string? PromoCode, decimal GstAmount, decimal DeliveryCharge, decimal TotalAmount,
    PaymentStatus PaymentStatus, decimal AmountPaid,
    DateTimeOffset CreatedAtUtc, DateTimeOffset? PromisedByUtc, DateTimeOffset? DeliveredAtUtc,
    DateTimeOffset? PickupScheduledAtUtc, List<OrderItemDetailDto> Items);
